### PR TITLE
progress indicator

### PR DESCRIPTION
**Pull request by SparshithNR**
Add the progress indicator to keep users informed about the current process status. 
[iteration 3]
progress

[iteration 2]
progress

[iteration 1]
progress

### CODE REVIEW

**Comment by stefanpenner** [comment]
The "new" version is nice for the eyeballs, but appears too not be the reality of what is happening. Rather prints a UI that is orthogonal to the progress and expected output.

The "old" version could likely use some improved verbiage, but represents reality a little better.

This is a good step, but we should iterate further.

File: lib/util.js [hunk 1]
```diff
@@ -63,9 +64,57 @@ function sortLibraries(a, b) {
   return a.type.localeCompare(b.type);
 }
 
+// spinner constant
```
Comment (stefanpenner) [inline lib/util.js:67]: let's avoid module level state, let's assume the algorithms in question can be re-run.

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: lib/util.js [hunk 2]
```diff
@@ -63,9 +64,57 @@ function sortLibraries(a, b) {
...
+let totalPackages = 1; // start from `1` as we have node policy tested all the time
+let spinner = 0;
+let processedCount = 0;
+let semVerLogged = false;
+function initSpinnerInfo(packagesCount, _spinner) {
```
Comment (stefanpenner) [inline lib/util.js:72]: Initing here, implies there is an implicit object at play, could we make it an explicit object instead? As they convey their intent clearly and consistently.

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: lib/util.js (hunk 1 above)
Comment (SparshithNR) [reply lib/util.js]: Moved this whole logic into a class so that we avoid the module level mutable state.

File: lib/util.js (hunk 2 above)
Comment (SparshithNR) [reply lib/util.js]: https://github.com/stefanpenner/supported/pull/8#discussion_r592557296

**Review by stefanpenner — APPROVED** [review]

**Pull request merged**